src/test/java/io/temporal/workflow/WorkflowTest.java

```diff
@@ -3995,7 +3996,10 @@ public void testWorkflowFailureNonRetryableFlag() {
     } catch (WorkflowException e) {
       assertTrue(e.getCause() instanceof ApplicationFailure);
       assertEquals("foo", ((ApplicationFailure) e.getCause()).getType());
-      assertEquals("details", ((ApplicationFailure) e.getCause()).getDetails().get(String.class));
+      assertEquals(
+          "details1", ((ApplicationFailure) e.getCause()).getDetails().get(0, String.class));
```

Comment:
and we still have a test that validates get without the index specified works fine?

Reply:
I checked and found at least two unit tests that call it.